docling_eval/evaluators/ocr/performance_calculator.py

```diff
@@ -0,0 +1,646 @@
+
+    return metrics_dict
+
+
+class ModelPerformanceCalculator:
+
+    def __init__(
+        self,
+        prediction_data: dict,
+        gt_data: dict,
+        add_space_between_merged_gt_words=True,
+        add_space_between_merged_prediction_words=True,
+        string_normalize_map=None,
+    ):
+        self.prediction_data = prediction_data
+        self.gt_data = gt_data
+        self.add_space_between_merged_gt_words = add_space_between_merged_gt_words
+        self.add_space_between_merged_prediction_words = (
+            add_space_between_merged_prediction_words
+        )
+        self.string_normalize_map = string_normalize_map if string_normalize_map else {}
+        self.prediction_words_original = self.prediction_data["images"][0]["words"]
+        self.gt_words_original = self.gt_data["images"][0]["words"]
+        self.__reset_matched_property()
+        self.gt_json_unified = copy.deepcopy(gt_data)
+        self.prediction_json_unified = copy.deepcopy(prediction_data)
+        self.gt_boxes_that_are_fn_after_refinement: list = []
+        self.prediction_boxes_that_are_fp_after_refinement: list = []
+        self.prediction_boxes_that_were_merged: list = []
+        self.gt_boxes_that_were_merged: list = []
+        self.fp_boxes = BoxesTypes([], [], [])
+        self.fn_boxes = BoxesTypes([], [], [])
+        self.__do_evaluation()
+
+    def __do_evaluation(self):
+        gt_to_prediction_boxes_map_val, prediction_to_gt_boxes_map_val = (
+            match_ground_truth_to_prediction_words(
+                self.gt_words_original, self.prediction_words_original
+            )
+        )
+        self.gt_to_prediction_boxes_map = gt_to_prediction_boxes_map_val
+        self.prediction_to_gt_boxes_map = prediction_to_gt_boxes_map_val
+        self.num_prediction_boxes_without_gt_intersection_val = sum(
+            [0 == len(v) for k, v in self.prediction_to_gt_boxes_map.items()]
+        )
+        self.num_gt_boxes_without_pred_intersection_val = sum(
+            [0 == len(v) for k, v in self.gt_to_prediction_boxes_map.items()]
+        )
+        self.__create_unified_words()
+
+    def num_prediction_boxes_without_gt_intersection(self):
+        return self.num_prediction_boxes_without_gt_intersection_val
+
+    def get_modified_jsons(self):
+        return self.gt_json_unified, self.prediction_json_unified
+
+    def __get_overlapped_gt_words(self, prediction_word):
+        box_key = box_to_key(prediction_word["location"])
+        if box_key in self.prediction_to_gt_boxes_map:
+            return self.prediction_to_gt_boxes_map[box_key]
+        return []
+
+    def __get_overlapped_prediction_words(self, gt_word):
+        gt_box_key = box_to_key(gt_word["location"])
+        if gt_box_key in self.gt_to_prediction_boxes_map:
+            return self.gt_to_prediction_boxes_map[gt_box_key]
+        return []
+
+    def __create_unified_words(self):
```

Comment:
very long function, please try to break them up into smaller functions, eg the parts starting with,

`for gt_word in self.gt_words_original:`, `for prediction_word in self.prediction_words_original:` and `for prediction_word, intersections in self.prediction_to_many_gt_boxes_map:` should be easy to factor out.